src/main/java/at/ac/tuwien/kr/alpha/solver/TrailAssignment.java

```diff
@@ -96,22 +100,25 @@ public void decreaseActivity() {
 	private boolean checksEnabled;
 	int replayCounter;
 
-	public TrailAssignment(AtomStore atomStore, boolean checksEnabled) {
+	public TrailAssignment(AtomStore atomStore, PhaseInitializerFactory.PhaseInitializer phaseInitializer, boolean checksEnabled) {
+		this.phaseInitializer = phaseInitializer;
 		this.checksEnabled = checksEnabled;
 		this.atomStore = atomStore;
 		this.values = new int[0];
 		this.strongDecisionLevels = new int[0];
 		this.impliedBy = new Antecedent[0];
 		this.callbackUponChange = new boolean[0];
+		this.phase = new boolean[0];
+		this.hasPhaseSet = new boolean[0];
 		this.trailIndicesOfDecisionLevels.add(0);
 		nextPositionInTrail = 0;
 		newAssignmentsIterator = 0;
 		newAssignmentsPositionInTrail = 0;
 		assignmentsForChoicePosition = 0;
 	}
 
-	public TrailAssignment(AtomStore atomStore) {
-		this(atomStore, false);
+	public TrailAssignment(AtomStore atomStore, PhaseInitializerFactory.PhaseInitializer phaseInitializer) {
```

Comment:
The argument `phaseInitializer` is not used.

Reply:
Done.